Automattic/wp-calypso
# Stats: Refactor shortcut matching

## Proposed Changes

Cleans up the logic used when matching the date range to a shortcut.

**Note:** Based on work done in #83170 so that should be merged first.

## Testing Instructions



1. Open Live branch to the **Stats → Traffic** page.
2. Enable date picker: `?flags=stats/date-control`
3. Manually enter some dates that should match shortcuts and confirm the shortcut is properly selected.

An easy way to test this is to pick a shortcut from the list, then manually change the start date by one day and update. Confirm the button label shows a full date range and no shortcut is selected. Then change the start date back and update. The label should be set back to the shortcut name and it should be selected in the list.

## Pre-merge Checklist
